lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/ICloudBlobClient.cs

```diff
@@ -46,6 +46,13 @@ internal interface ICloudBlobClient
         /// <returns>A reference to the blob container</returns>
         ICloudBlobContainer GetContainerReference(string containerName);
 
+        /// <summary>
+        /// Create the container specified in the uri if it does not exist already.
+        /// </summary>
+        /// <param name="uri">Uri of a blob</param>
+        /// <returns>Whether a new container was created or not</returns>
+        bool CreateContainerIfNotExists(Uri uri);
```

Comment:
Keep functions scoped to one thing. Much like GetContainerReference takes a containerName, so should CreateContainerIfNotExists. Otherwise, you are conflating two operations: 1. Extract the container name from a URI. 2. Creating a container if it doesn't exits.

Reply:
fixed